File: SPRING/Hospital_Management_system_angular/src/app/services/billing/billing-dashboard.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../../environments/environment';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class BillingDashboardService {
  private api = environment.apiUrl + 'billing-dashboard';
  constructor(private http: HttpClient) {}
  getSummary(): Observable<any> { return this.http.get<any>(`${this.api}/summary`); }
  getDailyRevenueChart(): Observable<any[]> { return this.http.get<any[]>(`${this.api}/daily-revenue-chart`); }
  getMonthlyRevenueChart(): Observable<any[]> { return this.http.get<any[]>(`${this.api}/monthly-revenue-chart`); }
  getDepartmentRevenue(): Observable<any[]> { return this.http.get<any[]>(`${this.api}/department-revenue`); }
  getPaymentMethodDistribution(): Observable<any[]> { return this.http.get<any[]>(`${this.api}/payment-methods`); }
  getRecentActivity(): Observable<any[]> { return this.http.get<any[]>(`${this.api}/recent-activity`); }
}

File: SPRING_BOOT/Modern_Hospital_Management_System/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/BillingDashboardService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import java.util.List;
import java.util.Map;

public interface BillingDashboardService {

    Map<String, Object> getDashboardSummary();

    List<Map<String, Object>> getDailyRevenueChart();

    List<Map<String, Object>> getMonthlyRevenueChart();

    List<Map<String, Object>> getDepartmentRevenue();

    List<Map<String, Object>> getPaymentMethodDistribution();

    List<Map<String, Object>> getRecentActivity();
}
